azavea/loam
# Enable running Loam from a CDN like unpkg

## Overview

This turned into a bit of a grab-bag, but it does two key things to improve developer experience:
- Enables running Loam from a CDN (e.g. unpkg). This closes #58 and will make it easier to run Loam from CodePen and similar.
- Creates a demo page that can be run by executing `yarn demo`. This provides a complete example of how to utilize loam in a simple webapp, and can be used as a test bed for features (in fact, it uses `unpkg` to load GDAL) or as a playground.

### Notes

- The way that this achieves the CDN loading is not ideal, for the reasons described in the comment in the code. I think I've done enough research to convince myself that this is the best way forward, but I would be very happy to have a discussion about the approach to double-check.

## Testing Instructions

 * Run `yarn install` to pick up dependency upgrades.
 * Running tests should all pass (CI is handling this, but you can follow the directions in the README to replicate)
 * Running `yarn demo` should result in a demo page being available on localhost:8080 -- you should be able to edit Loam source or the page source and change the behavior of the page.

## Checklist


Resolves #58
